File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.java
class Solution {
    public int maxAdjacentDistance(int[] nums) {
        int max = Math.abs(nums[nums.length - 1] - nums[0]);
        for(int i=0;i<nums.length-1;i++)
        {
            int d = Math.abs(nums[i+1] - nums[i]);
            if(d > max)
            max = d;
        }
        return max;
    }
}

File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.c
int maxAdjacentDistance(int* a, int n) {
    int i,j,c,d;
    c=0;
    j=0;
    for(i=0;i<n-1;i++)
    {
        d=abs(a[i]-a[i+1]);
        if(d>c)
        {
            c=d;
        }
    }
    j=abs(a[0]-a[n-1]);
    if(j>c)
    {
        c=j;
    }
        return c;
    }